numpyro/infer/inspect.py

```diff
@@ -364,16 +366,24 @@ def get_log_probs(sample):
             data=sample
         ):
             model(*model_args, **model_kwargs)
-        return {
-            name: site["fn"].log_prob(site["value"])
-            for name, site in tr.items()
-            if site["type"] == "sample"
-        }
+        # return {
+        #     name: site["fn"].log_prob(site["value"])
+        #     for name, site in tr.items()
+        #     if site["type"] == "sample"
+        # }
+        provenance_arrays = {}
+        for name, site in tr.items():
+            if site["type"] == "sample":
+                provenance_arrays[name] = site["fn"].log_prob(site["value"])
+            elif site["type"] == "deterministic":
+                provenance_arrays[name] = site["value"]
+        return provenance_arrays
 
     samples = {
         name: ProvenanceArray(site["value"], frozenset({name}))
         for name, site in trace.items()
         if (site["type"] == "sample" and not site["is_observed"])
+        # or site["type"] == "deterministic"
     }
 
```

Comment:
After getting `sample_sample` dict, we know which variables the deterministic site depends on, I guess we can (in most cases) tell which plates the site belongs to:
```
deterministic_sample = {name: v for name, v in sample_sample.items() if sample_dist[name] == "Deterministic"}
for name, deps in sample_sample.items():
    if sample_dist[name] == "Deterministic":
        for plate, samples in plate_samples.items():
            if len(set(deps) - set(samples)) == 0:
                samples.append(name)
                
```

Reply:
Sorry, the logic to decide which plates a deterministic site belongs to is more complicated. Let me think a bit more.

Reply:
I think we need to store plate info in the deterministic message. Let me add another comment for it.